substrate/frame/system/src/lib.rs

```diff
@@ -742,8 +742,6 @@ pub mod pallet {
 		#[pallet::call_index(8)]
 		#[pallet::weight(task.weight())]
 		pub fn do_task(origin: OriginFor<T>, task: T::RuntimeTask) -> DispatchResultWithPostInfo {
-			ensure_signed(origin)?;
-
 			if !task.is_valid() {
 				return Err(Error::<T>::InvalidTask.into())
```

Comment:
Is this defensive? I think the `validate_unsigned` should have prevented this?

Reply:
Yeah, `validate_unsigned` only allows those to be passed that meet the validity criteria. Anyone can still send the signed ones though, in which case they pay for execution.

Reply:
I see. In theory it would only need to check for validity in the signed case?  
That would make it slightly cheaper for unsigned ones. But okay, thanks for the explanation.